Plugin.cs

```diff
@@ -184,76 +184,97 @@ public void ReplacePlayer(Player playerToReplace)
             if (playerToReplace == null || playerToReplace.Position == null || !playerToReplace.IsAlive)
                 return;
 
-            var position = playerToReplace.Position;
             if (playerToReplace.IsInElevator())
                 return;
 
-            // regular spectator replacement
-            var spectators = Player.List.Where(p => p.Role.Type == RoleTypeId.Spectator);
+            // Regular spectator replacement
+            IEnumerable<Player> spectators = Player.List.Where(p => p.Role.Type == RoleTypeId.Spectator);
             if (spectators == null || spectators.IsEmpty())
                 return;
 
-            var spectator = spectators.RandomElement();
+            Player spectator = spectators.RandomElement();
 
             spectator.Role.Set(playerToReplace.Role.Type, SpawnReason.ForceClass, RoleSpawnFlags.UseSpawnpoint);
-            var roleName = $"<color={playerToReplace.Role.Color.ToHex()}>{playerToReplace.Role.Type.GetFullName()}</color>";
+            string roleName = $"<color={playerToReplace.Role.Color.ToHex()}>{playerToReplace.Role.Type.GetFullName()}</color>";
             foreach (CustomRole role in playerToReplace.GetCustomRoles())
             {
                 role.AddRole(spectator);
                 roleName = $"<color={playerToReplace.Role.Color.ToHex()}>{role.Name}</color>";
             }
+            
+            SetPlayerStats(playerToReplace, spectator);
+            spectator.Broadcast(10, $"<color=yellow>You have replaced an AFK player and become <color={playerToReplace.Role.Color.ToHex()}>{roleName}</color>.</color>");
+        }
+        
+        private void OnLeft(LeftEventArgs ev)
+        {
+            if(ev.Player.IsDead)
```

Comment:
Is this still defined after the player has left?

Reply:
I'm pretty sure, yes.